app/src/pages/inside/dashboardItemPage/modals/common/widgetInfoBlock/widgetInfoBlock.jsx

```diff
@@ -49,21 +49,33 @@ export class WidgetInfoBlock extends PureComponent {
 
   componentDidUpdate(prevProps) {
     const { filterIds = [], contentParameters = { widgetOptions: {} } } = this.props.widgetSettings;
+
     if (prevProps.activeWidget.id !== this.props.activeWidget.id) {
       this.resetPrevWidgetData();
       return;
     }
 
     const prevData = {
       filterIds: prevProps.widgetSettings.filterIds,
-      contentParameters: prevProps.widgetSettings.contentParameters,
+      contentParameters: prevProps.widgetSettings.contentParameters ?? {},
     };
 
     const newData = {
       filterIds,
       contentParameters,
     };
 
+    const {
+      widgetOptions: { includeSkipped: currentPassingRateOptions },
+    } = contentParameters;
+    const {
+      contentParameters: { widgetOptions: { includeSkipped: prevPassingRateOptions } = {} },
+    } = prevData;
+
+    if (currentPassingRateOptions !== prevPassingRateOptions) {
+      return;
+    }
```

Comment:
To prevent re-render on option change in this place : 
![screenshot_03](https://user-images.githubusercontent.com/84175555/198302516-973990c5-325c-41d9-9745-64f176eed8af.jpg)

Reply:
I guess it should be rerendered if there will be new values, isn't it?
I you want to stop rendering for case with unprovided required fields to show the real preview, I guess it should be fixed in different way.

Reply:
When we change value from All test, to Exclude skipped, this image is re-render, but we do not need it, nothing change in image appearance. And why it is a bad idea to use life cycle hook, to prevent unnecessary re-render?